client/admin/media/index.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import Gallery from 'react-grid-gallery';
+import { useListController, useQueryWithStore } from 'react-admin';
+import { useRef } from 'react';
+import { HOST } from '../App';
+
+const MediaGallery = props => {
+  const imagesInObject = useListController(props).data;
+  const fileInputRef = useRef(null);
+  const [file, setFile] = useState(null);
+  const imagesRaw = Object.values(imagesInObject);
+  const handleClick = event => {
+    fileInputRef.current.click();
+  };
+
+  const images = imagesRaw.map(image => ({
+    thumbnailWidth: 240,
+    thumbnailHeight: 240,
+    src: `${HOST}${image.relativeSrc}`,
+    thumbnail: `${HOST}${image.relativeSrc}`,
+  }));
+
+  const handleFileUpload = event => {
+    if (event.target.files[0]) {
+      setFile(event.target.files[0]);
+    }
+  };
+
+  return (
+    <div>
+      <button onClick={handleClick}>hello world</button>
+      <input
+        type="file"
+        onChange={handleFileUpload}
+        ref={fileInputRef}
+        style={{ display: 'none' }}
+      ></input>
+      <Gallery images={images} margin={4} rowHeight={240} />,
```

Comment:
it's probably more reliable to do gallery in react admin way 
checkout https://marmelab.com/react-admin-demo/#/products
and https://github.com/marmelab/react-admin/blob/master/examples/demo/src/products/GridList.js

Reply:
Actually, I have tried already to do that, you can checkout my code, it's not in a working state, though:
- media library folder - https://github.com/flatlify/flatlify/tree/media-library/client/admin/media-library
- https://github.com/flatlify/flatlify/blob/media-library/server/utils/common.js#L21 - My original intent was to avoid meta files and just recursively read all the files. And now I doubt if meta files are really necessary :|